lib/api_movie.dart

```diff
@@ -0,0 +1,101 @@
+
+import 'api_genres.dart';
+import 'genre.dart';
+import 'movie.dart';
+
+class ApiMovie {
+  final bool adult;
+  final double popularity;
+  final int id;
+  final int voteCount;
+  final String title;
+  final String originalTitle;
+  final bool video;
+  final List<dynamic> genres;
+  final String originalLanguage;
+  final String releaseDate;
+  final num voteAverage;
+  final String overview;
+  final String backdrop;
+  final String poster;
+
+  ApiMovie({
+    required this.adult,
+    required this.backdrop,
+    required this.genres,
+    required this.id,
+    required this.originalLanguage,
+    required this.originalTitle,
+    required this.overview,
+    required this.popularity,
+    required this.poster,
+    required this.releaseDate,
+    required this.title,
+    required this.video,
+    required this.voteAverage,
+    required this.voteCount,
+  });
+
+  factory ApiMovie.fromJson(Map<String, dynamic> json) {
+    return ApiMovie(
+      adult: json['adult'],
+      backdrop: json['backdrop_path'],
+      genres: json['genre_ids'],
+      id: json['id'],
+      originalLanguage: json['original_language'],
+      originalTitle: json['original_title'],
+      overview: json['overview'],
+      popularity: json['popularity'],
+      poster: json['poster_path'],
+      releaseDate: json['release_date'],
+      title: json['title'],
+      video: json['video'],
+      voteAverage: json['vote_average'],
+      voteCount: json['vote_count'],
+    );
+  }
+
+  static Movie mockMovie() {
+    return Movie(
+      adult: false,
+      backdrop:
+          'https://www.themoviedb.org/t/p/original/9n2tJBplPbgR2ca05hS5CKXwP2c.jpg',
+      genres: ApiGenre.mockGenres(),
+      id: 1,
+      originalLanguage: 'en',
+      originalTitle: 'The Super Mario Bros. Movie',
+      overview:
+          'While working underground to fix a water main, Brooklyn plumbers—and brothers—Mario and Luigi '
+          'are transported down a mysterious pipe and wander into a magical new world. But when the '
+          'brothers are separated, Mario embarks on an epic quest to find Luigi.',
+      popularity: 58,
+      poster:
+          'https://www.themoviedb.org/t/p/original/qNBAXBIQlnOThrVvA6mA2B5ggV6.jpg',
+      releaseDate: '2023-04-05',
+      title: 'The Super Mario Bros. Movie',
+      video: false,
+      voteAverage: 7.8,
+      voteCount: 0,
+    );
+  }
+
+  Movie toModel(List<Genre> genresList) {
+    return Movie(
+      adult: adult,
+      backdrop: '${strings.imageUrlPrefix}$backdrop',
+      genres: genresList.where((genre) => genres.contains(genre.id)).toList(),
+      id: id,
+      originalLanguage: originalLanguage,
+      originalTitle: originalTitle,
+      overview: overview,
+      popularity: popularity,
+      poster: '${strings.imageUrlPrefix}$poster',
+      releaseDate: releaseDate,
+      title: title,
+      video: video,
+      voteAverage: voteAverage,
+      voteCount: voteCount,
+    );
+  }
+}
```

Comment:
Move the configuration to get the full backgrop and poster path to a single get methods like:
`
String get fullBackgropPath => '${strings.imageUrlPrefix}$backdrop';
String get fullPosterPath => '${strings.imageUrlPrefix}$poster';
`
And also the logic to parse the genres could be on the UI side, here you only need to get the ids